lib/plug/parsers.ex

```diff
@@ -200,23 +200,21 @@ defmodule Plug.Parsers do
       nil ->
         %{conn | body_params: %{}}
     end
+    merge_params(conn)
   end
 
   def call(%Conn{body_params: %Plug.Conn.Unfetched{}} = conn, _opts) do
-    conn = Conn.fetch_query_params(conn)
-    %{conn | body_params: %{}}
+    conn |> Conn.fetch_query_params |> merge_params
   end
 
   def call(%Conn{} = conn, _opts) do
-    Conn.fetch_query_params(conn)
+    conn |> Conn.fetch_query_params |> merge_params
```

Comment:
I don't think we need to call merge_params here. fetch_query_params is going to guarantee the proper ordering.